pkg/chart/repo/handlers.go

```diff
@@ -22,6 +22,10 @@ func ResolveChartVersionFunc(c *Catalog) ChartVersionResolver {
 
 		if _, err := repo.RefreshIndex(); err != nil {
 			glog.Warningf("failed to refresh repo[%s] index: %s", chartspec.RepoURL, err)
+			switch err.(type) {
+			case errors.NoCachedChartRepoIndexError:
+				return nil, err
```

Comment:
I suggest adding `default:` statement (even an empty one if a fall through is meant here) so the alternative action is explicitly visible.